src/typeahead/typeahead.js

```diff
@@ -332,12 +338,20 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position'])
             return candidateViewValue!== emptyViewValue ? candidateViewValue : modelValue;
           }
         });
-
+        //method for clear blutton
+        scope.clearModel = function(){
+           $setModelValue(originalScope, '');
+           scope.selectedM = false;  
+           scope.$digest();            
+        }; 
+        
         scope.select = function(activeIdx) {
           //called from within the $digest() cycle
           var locals = {};
           var model, item;
-
+          //indicator that some choise selected for clear button
+		  scope.selectedM = true;
```

Comment:
Hi Wesley !
Was very pleased to get this email from you
Trying to make the tests as soon as possible.

Thanks


On Tue, Aug 16, 2016 at 6:57 PM, Wesley Cho <notifications@github.com>
wrote:

> In src/typeahead/typeahead.js
> <https://github.com/angular-ui/bootstrap/pull/4483#discussion_r74963983>:
>
> >          scope.select = function(activeIdx) {
> >            //called from within the $digest() cycle
> >            var locals = {};
> >            var model, item;
> > -
> > +          //indicator that some choise selected for clear button
> > +		  scope.selectedM = true;
>
> Fix indentation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular-ui/bootstrap/pull/4483/files/63ec31dd8488198b4f6168c80e965c979442c900#r74963983>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABRmHniBdfU0TZfo97Qm2HMYXZiFGyrLks5qgd3ZgaJpZM4GEd5X>
> .
>



-- 
http://gs500coder.blogspot.com